packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/http-private.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  HttpConnectionType,
+  HttpIntegrationType,
+  HttpRouteIntegrationBindOptions,
+  HttpRouteIntegrationConfig,
+  IHttpRouteIntegration,
+  PayloadFormatVersion,
+  HttpMethod,
+  IVpcLink,
+} from '@aws-cdk/aws-apigatewayv2';
+import * as ec2 from '@aws-cdk/aws-ec2';
+
+/**
+ * Base properties for private integration
+ */
+export interface BasePrivateIntegrationProps {
+  /**
+   * The vpc link to be used for the private integration
+   *
+   * @default - a new VpcLink is created
+   */
+  readonly vpcLink?: IVpcLink;
+
+  /**
+   * The HTTP method that must be used to invoke the underlying HTTP proxy.
+   * @default HttpMethod.ANY
+   */
+  readonly method?: HttpMethod;
+}
+
+
+/**
+ * Properties to initialize a new `HttpPrivateIntegration`.
+ */
+export interface HttpPrivateIntegrationProps extends BasePrivateIntegrationProps {
+  /**
+   * The vpc for which the VpcLink needs to be created
+   *
+   * @default - undefined
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * The ARN of the target resource.
+   * One of ApplicationLoadBalancerListener, NetworkLoadBalancerListener or CloudMapDiscoveryService
+   */
+  readonly targetArn: string;
+}
+
+/**
+ * The HTTP Private integration resource for HTTP API
+ */
+export class HttpPrivateIntegration implements IHttpRouteIntegration {
+  constructor(private readonly props: HttpPrivateIntegrationProps) {
+  }
+
+  public bind(options: HttpRouteIntegrationBindOptions): HttpRouteIntegrationConfig {
+    let vpcLink = this.props.vpcLink;
+    if (!vpcLink) {
+      if (!this.props.vpc) {
+        throw new Error('One of vpcLink or vpc should be provided for private integration');
+      }
```

Comment:
error when both are provided and they don't point to the same vpc

Reply:
Is that validation required? We are currently ignoring the passed vpc if the vpcLink is supplied. Basically vpcLink trumps the vpc.

Also, the vpcLink in v2 can point to multiple VPCs in theory (since it takes subnets and securityGroups). In this case it might not make sense to compare the VPCs.

Thoughts?

Reply:
Accepting a property and not using it is usually confusing. "Why was my vpc not used?" would be a complaint I expect to hear.

How about just failing if both are presented? It'll make the requirement that only one should be passed clear.

Reply:
Yeah, sounds good.

Reply:
This should not be required anymore since we are not accepting vpc as a prop (it is passed around internally by integrations)

Reply:
How come? 
In the case of `HttpAlbIntegration`, we accept an ALB and a VPCLink. If the ALB is in a different vpc from the vpc link, it's useful to error, no?

Reply:
Oh that way. 

But in that case the intent was explicit to use the passed vpcLink right? We are not explicitly taking the vpc as a prop. We are inferring.

Also, as mentioned before, the vpcLink in v2 is not bound to just one vpc in theory. It revolves around subnets and securityGroups. We have just provided a convenience to pass a vpc to create the vpcLink. So comparing the vpc of vpcLink and ALB listener might not be ideal.

Thoughts?

Reply:
I see. Ok, let's leave it as is for now. See how this pans out.